tests/smoke-tests/test-dynamic-fees.ts

```diff
@@ -305,44 +321,68 @@ describeSmokeSuite(
+        .map(({ blockNum, matchedEvents, filteredTxnEvents, transactionStatuses, receipts }) => {
+          const fees = matchedEvents
             .filter((a) => isEthereumTxn(blockNum, a.phase.asApplyExtrinsic.toNumber()))
-            .map(({ event }, index) => {
+            .map(({ event }) => {
               const info = event.data[0] as DispatchInfo;
-              const fee = info.weight as unknown as SpWeightsWeightV2Weight;
-              return fee;
+              const fee = info.weight as SpWeightsWeightV2Weight;
+              return fee.refTime;
             });
-          const ethExtGas = receipts.map((item) => {
-            switch (true) {
-              case item.isEip1559:
-                return item.asEip1559.usedGas;
-              case item.isEip2930:
-                return item.asEip2930.usedGas;
-              case item.isLegacy:
-                return item.asLegacy.usedGas;
-              default:
-                throw new Error("Update test to include new transaction type");
-            }
-          });
-          return { blockNum, ethExtFees, ethExtGas, ethersTransactionsFees };
+
+          const gasUsed: u256[] = filteredTxnEvents
+            .map((a) => {
+              if (isEthereumTxn(blockNum, a.phase.asApplyExtrinsic.toNumber())) {
+                const txnHash = (a.event.data as any).transactionHash;
+                const index = transactionStatuses.findIndex((a) => a.transactionHash.eq(txnHash));
+                // Gas used is cumulative measure, so we have to derive individuals
+                const gasUsed =
+                  index === 0
+                    ? extractGasAmount(receipts[index])
+                    : extractGasAmount(receipts[index]).sub(extractGasAmount(receipts[index - 1]));
+                return gasUsed;
+              } else {
+                return [];
+              }
+            })
+            .flatMap((a) => a);
+
+          expect(fees.length, "More eth reciepts than expected, this test needs fixing").to.equal(
+            gasUsed.length
+          );
+
+          const estimatedFees = gasUsed.map((a) => a.mul(new BN(WEIGHT_PER_GAS.toString())));
+
+          const matchedAmounts = estimatedFees
+            .map((a, index) => a.eq(fees[index].toBn()).valueOf())
+            .reduce((curr, arr) => curr && arr, true);
+
+          return { blockNum, fees, gasUsed, estimatedFees, matchedAmounts };
         })
-        .filter((a) => a.ethExtFees.length > 0)
-        .filter((item) => {
-          const match = item.ethExtFees
-            .map((subitem, index) =>
-              subitem.refTime.eq(item.ethersTransactionsFees[index].mul(WEIGHT_PER_GAS.toString()))
-            )
-            .reduce((acc, curr) => acc && curr, true);
-
-          return !match;
-        });
+        .filter((a) => a.matchedAmounts !== true);
 
-      failures.forEach(({ blockNum, ethExtFees, ethExtGas }) => {
+      failures.forEach((a) => {
```

Comment:
I strongly suggest to forbid one letter variables, if you can provide a better name it would be great

Reply:
Done